File: HomeBudget.Data/Entities/SavingsGoal.cs
namespace HomeBudget.Data.Entities;

public class SavingsGoal
{
    public int Id { get; set; }
    public string Name { get; set; } = string.Empty;
    public decimal TargetAmount { get; set; }
    public decimal CurrentAmount { get; set; } = 0;
    public DateTime? Deadline { get; set; }
    public string Icon { get; set; } = "piggy-bank";
    public string Color { get; set; } = "#10B981";
    public bool IsCompleted { get; set; } = false;

    public int HouseholdId { get; set; }
    public DateTime CreatedAt { get; set; } = DateTime.UtcNow;

    public Household Household { get; set; } = null!;
}

File: HomeBudgetMobile/src/models/savingsGoal.ts
export interface SavingsGoalDto {
  id: number;
  name: string;
  targetAmount: number;
  currentAmount: number;
  percentage: number;
  deadline: string | null;
  icon: string;
  color: string;
  isCompleted: boolean;
}

export interface CreateSavingsGoalDto {
  name: string;
  targetAmount: number;
  deadline?: string;
  icon: string;
  color: string;
  householdId: number;
}

export interface UpdateSavingsGoalDto {
  id: number;
  name: string;
  targetAmount: number;
  currentAmount: number;
  deadline?: string | null;
  icon: string;
  color: string;
  isCompleted: boolean;
}
